Lib/test/test_typing.py

```diff
@@ -5182,6 +5182,18 @@ class C(B[int]):
                 x = pickle.loads(z)
                 self.assertEqual(s, x)
 
+        # Test ParamSpec args and kwargs
+        global PP
+        PP = ParamSpec('PP')
+        for thing in [PP.args, PP.kwargs]:
+            for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+                with self.subTest(thing=thing, proto=proto):
+                    self.assertEqual(
+                        pickle.loads(pickle.dumps(thing, proto)),
+                        thing,
+                    )
+        del PP
```

Comment:
should this be some sort of context manager so that `PP` is `del`'d from the global namespace even if an exception in the `for` loop leads to the test terminating early?

should we also `del` the `C` name, which is declared as `global` on the first line of this function?

Reply:
There are multiple places where this pattern is used. I think that it should be fine, because the only problem that I can think of is that `PP` will be reported as changed by `refleak`. So, not a big deal in case of a failure.

But, this can be refactored in the future, if desired.